### PR TITLE
add focal_loss in hungarian_assigner.py

### PR DESCRIPTION
**Pull request by kellenf**
some need  to review, this is a base_code

### CODE REVIEW

File: mmdet/core/bbox/assigners/hungarian_assigner.py [hunk 1]
```diff
@@ -42,13 +42,16 @@ def __init__(self,
                  bbox_weight=1.,
                  iou_weight=1.,
                  iou_calculator=dict(type='BboxOverlaps2D'),
-                 iou_mode='giou'):
+                 iou_mode='giou',
+                 focal_loss=dict(alpha=0.25, gamma=2.0),
```
Comment (ZwwWayne) [inline mmdet/core/bbox/assigners/hungarian_assigner.py:46]: comma is unnecessary.

File: mmdet/core/bbox/assigners/hungarian_assigner.py [hunk 2]
```diff
@@ -116,14 +119,26 @@ def assign(self,
...
+        else:
+            cls_score = cls_pred.softmax(-1)
+            cls_cost = -cls_score[:, gt_labels]  # [num_bboxes, num_gt]
 
         # regression L1 cost
         img_h, img_w, _ = img_meta['img_shape']
         factor = torch.Tensor([img_w, img_h, img_w,
                                img_h]).unsqueeze(0).to(gt_bboxes.device)
         gt_bboxes_normalized = gt_bboxes / factor
+        # MODYFY: xyxy p1-norm in sparse-RCNN CODE which need to review
```
Comment (ZwwWayne) [inline mmdet/core/bbox/assigners/hungarian_assigner.py:141]: We may also use build loss for bbox distance.

File: mmdet/core/bbox/assigners/hungarian_assigner.py (hunk 1 above)
Comment (kellenf) [reply mmdet/core/bbox/assigners/hungarian_assigner.py]: Haved updated, use `build_match_cost` to build all loss, please review

File: mmdet/core/bbox/assigners/hungarian_assigner.py (hunk 2 above)
Comment (kellenf) [reply mmdet/core/bbox/assigners/hungarian_assigner.py]: updated

**Pull request closed without merging**